### PR TITLE
Made at least some kind of modularity for plugins

### PR DESCRIPTION
**Pull request by Muinez**
I added `shared` folder containing libraries to be shared by plugins

```
counterstrikesharp
|-- shared
|   |-- Utils
|       |-- Utils.deps.json
|       |-- Utils.dll
|       |-- Utils.pdb
```
both core and module plugin should be dependent on shared library (in this case, `Utils`)

example core plugin:
```
public class TestPlugin : BasePlugin
{
    public override string ModuleName => "TestPlugin";
    public override string ModuleVersion => "1.0.0";

    public TestPlugin()
    {
        AddApi<ITestApi>(new TestApi());
    }
}
    
public class TestApi : ITestApi
{
    public void PrintSomething()
    {
        Console.WriteLine("12345");
    }
}
```

example module plugin:
```
public class TestModule : BasePlugin
{
    public override string ModuleName => "TestModule";
    public override string ModuleVersion => "1.0.0";

    public override void Load(bool hotReload)
    {
        var api = GetApi<ITestApi>()!;

        api.PrintSomething();
    }
}
```

example shared library:
```
public interface ITestApi
{
    public void PrintSomething();
}
```

### CODE REVIEW

**Comment by KillStr3aK** [comment]
this is under discuss in https://github.com/roflmuffin/CounterStrikeSharp/issues/71

File: managed/CounterStrikeSharp.API/Core/GlobalContext.cs [hunk 1]
```diff
@@ -19,27 +19,71 @@
...
+                }
+                catch
+                {
+                    var assembly = _sharedAssemblies.First(a => a.name.Name == name.Name);
+                    return assembly.assembly;
+                }
+            };
+
+            LoadSharedLibs();
+        }
```
Comment (KillStr3aK) [inline managed/CounterStrikeSharp.API/Core/GlobalContext.cs:67]: `McMaster` has its own implementation of shared libraries so I'd say this is needless

**Comment by laper32** [comment]
please see #71

File: managed/CounterStrikeSharp.API/Core/GlobalContext.cs (hunk 1 above)
Comment (Muinez) [reply managed/CounterStrikeSharp.API/Core/GlobalContext.cs:67]: > `McMaster` has its own implementation of shared libraries so I'd say this is needless

what exactly are you talking about?

Comment (frederikstonge) [reply managed/CounterStrikeSharp.API/Core/GlobalContext.cs:67]: > > `McMaster` has its own implementation of shared libraries so I'd say this is needless
> 
> what exactly are you talking about?

https://www.nuget.org/packages/McMaster.NETCore.Plugins

They use this package to load plugin assemblies.

**Comment by partiusfabaa** [comment]
I don't think it can be done any better. You have to add

**Comment by KillStr3aK** [comment]
> I don't think it can be done any better. You have to add

We already have "plans" about the implementation

**Comment by partiusfabaa** [comment]
> > I don't think it can be done any better. You have to add
> 
> We already have "plans" about the implementation

I think its already a good option, so why do anything else if its already there?

**Comment by roflmuffin** [comment]
Hi @Muinez, would you be able to review #253 and see if this solves the use case for this PR?

**Pull request closed without merging**